crates/swc_ecma_transforms_compat/src/es2017/async_to_generator.rs

```diff
@@ -342,76 +345,76 @@ impl Fold for Actual {
                     .as_callee(),
                     args: vec![],
                     type_args: Default::default(),
-                })
+                });
             }
 
-            _ => expr,
+            _ => {}
         }
     }
 
-    fn fold_fn_decl(&mut self, f: FnDecl) -> FnDecl {
-        let f = f.fold_children_with(self);
+    fn visit_mut_fn_decl(&mut self, f: &mut FnDecl) {
+        f.visit_mut_children_with(self);
         if !f.function.is_async {
-            return f;
+            return;
         }
 
-        let function = self.fold_fn(Some(f.ident.clone()), f.function, true);
-        FnDecl {
-            ident: f.ident,
-            function,
+        self.visit_fn(Some(f.ident.clone()), &mut f.function, true);
+        *f = FnDecl {
+            ident: f.ident.take(),
```

Comment:
Assigning to `f.function` would be better than using `ident.take()`.

Reply:
Actually I realized f.function itself is visited with mut, reassign it again was not needed.